source/unix/openssl_platform_init.c

```diff
@@ -48,6 +69,25 @@ extern int HMAC_Update(HMAC_CTX *, const unsigned char *, size_t) __attribute__(
 extern int HMAC_Final(HMAC_CTX *, unsigned char *, unsigned int *) __attribute__((weak, used));
 extern int HMAC_Init_ex(HMAC_CTX *, const void *, int, const EVP_MD *, ENGINE *) __attribute__((weak, used));
 
+static void s_hmac_ctx_reset_openssl(HMAC_CTX *ctx) {
+    AWS_PRECONDITION(ctx);
+
+    int (*reset_ptr)(HMAC_CTX *) = (int (*)(HMAC_CTX *))g_aws_openssl_hmac_ctx_table->impl.reset_fn;
+    reset_ptr(ctx);
```

Comment:
add comment like
`/* openssl returns int, but other implementations don't */`

also, we should probably AWS_FATAL_ASSERT that it returns 0